pika/adapters/blocking_connection.py

```diff
@@ -1752,7 +1753,7 @@ def start_consuming(self):
 
         # Process events as long as consumers exist on this channel
         while self._consumer_infos:
-            self.connection.process_data_events(time_limit=None)
+            self.connection.process_data_events(time_limit=time_limit)
```

Comment:
How does this help it not wait forever? This won't help the while loop exit, will it?